File: templates/python/code-carl/ch05_stack/t2_lc0225_my_stack.py
"""LeetCode 225. Implement Stack using Queues
https://leetcode.cn/problems/implement-stack-using-queues/
"""


class MyStack:
    def __init__(self) -> None:
        pass

    def push(self, x: int) -> None:
        pass

    def pop(self) -> int:
        pass

    def top(self) -> int:
        pass

    def empty(self) -> bool:
        pass


if __name__ == "__main__":
    stack = MyStack()
    stack.push(1)
    stack.push(2)
    assert stack.top() == 2
    assert stack.pop() == 2
    assert stack.empty() is False
    assert stack.pop() == 1
    assert stack.empty() is True
    print("All tests passed.")

File: solutions/code-carl/ch05_stack/T2_LC0225_MyStack.java
/**
 * Practice template for:
 * - LeetCode 225. Implement Stack using Queues
 *   https://leetcode.cn/problems/implement-stack-using-queues/
 */
import java.util.ArrayDeque;
import java.util.Deque;

class T2_LC0225_MyStack {
    Deque<Integer> deque;

    public T2_LC0225_MyStack() {
        this.deque = new ArrayDeque<>();
    }

    public void push(int x) {
        deque.offerFirst(x);
    }

    public int pop() {
        return deque.pollFirst();
    }

    public int top() {
        return deque.peekFirst();
    }

    public boolean empty() {
        return deque.isEmpty();
    }
}
